api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java

```diff
@@ -42,14 +43,20 @@ public void shouldSaveClaimant() {
         claimService.createClaim(claim);
 
         //then
-        verify(claimantRepository).save(claim.getClaimant());
+        Claimant expectedClaimant = claimant.toBuilder().eligibilityStatus(EligibilityStatus.ELIGIBLE).build();
+        verify(claimantRepository).save(expectedClaimant);
         verify(client).checkEligibility(claimant);
     }
 
     @Test
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
```

Comment:
I would personally disable this rule as it's in the Controversial category and seems to cause a lot of false positives.